src/core/cpu/arm_interpret.cpp

```diff
@@ -886,6 +892,53 @@ uint32_t load_store_shift_reg(ARM_CPU& cpu, uint32_t instr)
     return reg;
 }
 
+void arm_sel(ARM_CPU &cpu, uint32_t instr)
+{
+    int reg1 = (instr >> 16) & 0xF;
+    int dest = (instr >> 12) & 0xF;
+    int reg2 = instr & 0xF;
+
+    uint32_t source1 = cpu.get_register(reg1);
+    uint32_t source2 = cpu.get_register(reg2);
+
+    uint32_t dest_word = 0;
+
+    const PSR_Flags* cpsr = cpu.get_CPSR();
+    for (int i = 0; i < 4; i++)
+    {
+        if (cpsr->ge[i])
+            dest_word |= ((source1 >> (i * 8)) & 0xFF) << (i * 8);
+        else
+            dest_word |= ((source2 >> (i * 8)) & 0xFF) << (i * 8);
+    }
+
+    cpu.set_register(dest, dest_word);
+}
+
+void arm_uadd8(ARM_CPU &cpu, uint32_t instr)
+{
+    int reg1 = (instr >> 16) & 0xF;
+    int dest = (instr >> 12) & 0xF;
+    int reg2 = instr & 0xF;
+
+    uint32_t source1 = cpu.get_register(reg1);
+    uint32_t source2 = cpu.get_register(reg2);
+
+    uint32_t dest_word = 0;
+
+    PSR_Flags* cpsr = cpu.get_CPSR();
+    for (int i = 0; i < 4; i++)
+    {
+        uint16_t op1 = (source1 >> (i * 8)) & 0xFF;
+        uint16_t op2 = (source2 >> (i * 8)) & 0xFF;
+        uint16_t res = op1 + op2;
+        cpsr->ge[i] = (res & 0xFF00 != 0);
```

Comment:
This should be `(res & 0xFF00) != 0`, or better yet, `res > 0xFF`. != has higher precedence than &, so the actual operation with your code becomes `res & 1` as `0xFF00 != 0` is always true.

Reply:
Shame on me for not enabling `-Wall -Wextra` and reading the messages on that one